docs/spec/spec.md

```diff
@@ -0,0 +1,322 @@
+
+Users can invoke it as shown below:
+
+```ballerina
+public function main() returns error? {
+    xml iDoc = check io:fileReadXml("resources/sample.xml");
+    check jcoClient->sendIDoc(iDoc);
+    io:println("IDoc sent successfully.");
+}
+```
+
+#### 2.1.3 Data Type Mapping Between Ballerina and SAP JCo
+
+The following table maps Ballerina data types to their corresponding SAP JCo types:
+
+| Ballerina Type                        | SAP JCo Type   | SAP Data Type         |
+| ------------------------------------- | -------------- | --------------------- |
+| string                                | char           | CHAR                  |
+| string                                | char[]         | STRING, SSTRING       |
+| time:Date                             | Date           | DATE                  |
+| time:TimeOfDay                        | Time           | TIME                  |
+| int                                   | short          | INT1                  |
+| int                                   | int            | INT2                  |
+| int                                   | long           | INT4                  |
+| float                                 | float          | FLOAT                 |
+| decimal                               | double         | DEC, QUAN, CURR, PREC |
+| byte                                  | byte           | INT1                  |
+| byte[]                                | byte[]         | RAW, LRAW, BYTE       |
+| decimal                               | BigDecimal     | DEC, QUAN, CURR       |
+| int                                   | BigInteger     | NUM                   |
+| record (with/without rest fields)     | JCoStructure   | STRUCTURE             |
+| record[] (with/without rest fields)   | JCoTable       | TABLE                 |
+
+### 2.2 Listener
+
+#### 2.2.1 Configurations
+
+The Listener supports two types of configurations, `ServerConfig` and `AdvancedConfig`, to facilitate communication with the SAP system by receiving IDocs and handling events.
+
+##### 2.2.1.1 ServerConfig
+
+The `ServerConfig` type represents the configuration details needed to create an IDoc connection.
+
+```ballerina
+public type ServerConfig record {
+    # The gateway host (jco.server.gwhost)
+    string gwhost;
+    # The gateway service (jco.server.gwserv)
+    string gwserv;
+    # The program ID (jco.server.progid)
+    string progid;
+};
+```
+
+**Example Configuration:**
+
+```ballerina
+configurable jco:ServerConfig configs = ?;
+listener jco:Listener idocListener = new (configs);
+```
+
+Config.toml file:
+
+```toml
+[configs]
+gwhost = "gwhost"
+gwserv = "sapgw00"
+progid = "progid"
+```
+
+##### 2.2.1.2 AdvancedConfig
```

Comment:
If this is the same thing we defined previously, no need to redefine.

Reply:
Yes, but in this case, 'That' is used to accept a different set of values from the Client. The initial thought was that we need to demonstrate this using an example, even for the developer, since it plays a critical role in certain SAP ECC scenarios. In some versions of SAP ECC, this can be used to create a Destination (which happens only if specific configurations are provided), where a client and a listener are initiated in the background.

Reply:
In that case this should have been something like this IMO;

#### 1. Using `AdvanceConfig` as field in the `ServerConfig`

```ballerina
public type ServerConfig record {
    # The gateway host (jco.server.gwhost)
    string gwhost;
    # The gateway service (jco.server.gwserv)
    string gwserv;
    # The program ID (jco.server.progid)
    string progid;
    # doc comment.....
    AdvancedConfig advanceConfig?;
};
```

#### 2. By opening the `ServerConfig` for `string` fields

Anyway `ServerConfig` is an open-record, this is always allowed.

Both the above options assumes that `gwhost`, `gwserv`, and `progid` are required fields in both regular and advance cases. 

Any reason to not to do that ? (again this is more like an issue with the design, but better to clarify it at this point)

Reply:
We followed this approach as it is much more user-friendly for both new and existing users.

Existing users already have a `.jcoDestination` file that contains a set of configurations like the following:

```
#Standard destination
jco.client.client=MANDANT
jco.client.ashost=APPLICATION_SERVER_HOST
jco.client.sysnr=INSTANZNUMMER
jco.client.r3name=SYSTEMID
jco.client.saprouter=SAP_ROUTER_STRING
jco.client.lang=DE
jco.server.mshost=
jco.client.user=USERNAME
jco.client.passwd=PASSWORD
jco.server.gwhost=GWHOST.COM
jco.server.gwserv=SAPGW00
jco.server.progid=PROGRAMID
jco.server.connection_count=1
jco.server.repository_destination=POOL2
```

This can easily be represented as a Ballerina map, which is why we introduced the `AdvancedConfig` record.

On the other hand, new users (using the later version of the SDK) will only need to obtain the following information from their SAP system administrator to create a listener. which reduces the cognitive load on the new users.

```
gwhost = GWHOST.COM
gwserv = SAPGW00
progid = PROGRAMID
```

However, the method suggested in '2. Opening the ServerConfig for string fields' is also a good option, and we will discuss it in future improvement planning meetings.